source/base/polynomials_nedelec_sz.cc

```diff
@@ -0,0 +1,188 @@
+
+
+#include <deal.II/base/polynomials_nedelec_sz.h>
+
+DEAL_II_NAMESPACE_OPEN
+namespace
+{
+  Threads::Mutex coefficients_lock;
+  
+}
+
+// Reserve space for polynomials up to degree 19.
+std::vector<std_cxx11::shared_ptr<const std::vector<double> > >
+integratedLegendreSZ::recursive_coefficients(20);
+
+std::vector<std_cxx11::shared_ptr<const std::vector<double> > >
+integratedLegendreSZ::shifted_coefficients(20);
+
+
+integratedLegendreSZ::integratedLegendreSZ (const unsigned int k)
+:
+Polynomials::Polynomial<double> (get_coefficients(k))
+{}
+
+void
+integratedLegendreSZ::compute_coefficients (const unsigned int k_)
+{
+  // NOT REQUIRED AT THE MOMENT:
+   
+  // make sure we call the
+  // Polynomial::shift function
+  // only with an argument with
+  // which it will not crash the
+  // compiler
+// #ifdef DEAL_II_LONG_DOUBLE_LOOP_BUG
+//     typedef double SHIFT_TYPE;
+// #else
+//     typedef long double SHIFT_TYPE;
+// #endif
+
+  unsigned int k = k_;
+
+  // first make sure that no other
+  // thread intercepts the
+  // operation of this function;
+  // for this, acquire the lock
+  // until we quit this function
+  Threads::Mutex::ScopedLock lock(coefficients_lock);
+
+  // The first 2 coefficients are hard-coded
+  if (k==0)
+    k=1;
+  // check: does the information
+  // already exist?
+  if ((recursive_coefficients.size() < k+1) ||
+    ((recursive_coefficients.size() >= k+1) &&
+    (recursive_coefficients[k] ==
+    std_cxx11::shared_ptr<const std::vector<double> >())))
+    // no, then generate the
+    // respective coefficients
+  {
+    // make sure that there is enough space in the array for the coefficients,
+    // so we have to resize it to size k+1
+    
+    // but it's more complicated than that: we call this function recursively, so if we simply
+    // resize it to k+1 here, then compute the coefficients for degree k-1 by calling this
+    // function recursively, then it will reset the size to k -- not enough for what we want to do below. the
+    // solution therefore is to only resize the size if we are going to *increase* it
+    if (recursive_coefficients.size() < k+1)
+    {
+      recursive_coefficients.resize (k+1);
+    }
+    if (k<=1)
+    {
+      // create coefficients vectors for k=0 and k=1
+      //
+      // allocate the respective later assign it to the coefficients array to make it const
+      std::vector<double> *c0 = new std::vector<double>(1);
+      (*c0)[0] = -1.;
+      
+      std::vector<double> *c1 = new std::vector<double>(2);
+      (*c1)[0] = 0.;
+      (*c1)[1] = 1.;
+      
+      // now make these arrays const. use shared_ptr for recursive_coefficients because
+      // that avoids a memory leak that would appear if we used plain pointers.
+      recursive_coefficients[0] =
+      std_cxx11::shared_ptr<const std::vector<double> >(c0);
+      recursive_coefficients[1] =
+      std_cxx11::shared_ptr<const std::vector<double> >(c1);
```

Comment:
since you started your patch, we've moved to require C++11, so you can just use `std::shared_ptr`.